yt_dlp/extractor/nytimes.py

```diff
@@ -124,139 +88,348 @@ class NYTimesIE(NYTimesBaseIE):
     }]
 
-    def _extract_podcast_from_json(self, json, page_id, webpage):
-        podcast_audio = self._parse_json(
-            json, page_id, transform_source=js_to_json)
+    def _extract_content_from_block(self, block):
+        details = traverse_obj(block, {
+            'id': ('sourceId', {str}),
+            'uploader': ('bylines', ..., 'renderedRepresentation', {str}),
+            'duration': (('duration', 'length', ...), {int_or_none}),
+            'timestamp': ('firstPublished', {parse_iso8601}),
+            'series': ('podcastSeries', {str}),
+        }, get_all=False)
+
+        formats, subtitles = self._extract_media_from_json(details.get('id'), block)
 
-        audio_data = podcast_audio['data']
-        track = audio_data['track']
+        # audio articles will have an url and no formats
+        url = traverse_obj(block, ('fileUrl', {url_or_none}))
+        if not formats and url:
+            formats.append({'url': url, 'vcodec': 'none'})
 
-        episode_title = track['title']
-        video_url = track['source']
+        return {
+            **details,
+            'formats': formats,
+            'thumbnails': self._extract_thumbnails(traverse_obj(
+                block, ('promotionalMedia', 'crops', ..., 'renditions', ...))),
+            'subtitles': subtitles
+        }
 
-        description = track.get('description') or self._html_search_meta(
-            ['og:description', 'twitter:description'], webpage)
+    def _real_extract(self, url):
+        page_id = self._match_id(url)
+        webpage = self._download_webpage(url, page_id)
+        art_json = self._search_json(
+            r'window\.__preloadedData\s*=', webpage, 'media details', page_id,
+            transform_source=lambda x: x.replace('undefined', 'null'))['initialData']['data']['article']
+
+        common_info = {
+            'title': remove_end(self._html_extract_title(webpage), ' - The New York Times'),
+            'description': traverse_obj(art_json, (
+                'sprinkledBody', 'content', ..., 'summary', 'content', ..., 'text', {str}),
+                get_all=False) or self._html_search_meta(['og:description', 'twitter:description'], webpage),
+            'timestamp': traverse_obj(art_json, ('firstPublished', {parse_iso8601})),
+            'creator': ', '.join(
+                traverse_obj(art_json, ('bylines', ..., 'creators', ..., 'displayName'))),  # TODO: change to 'creators' (list)
+            'thumbnails': self._extract_thumbnails(traverse_obj(
+                art_json, ('promotionalMedia', 'assetCrops', ..., 'renditions', ...))),
+        }
 
-        podcast_title = audio_data.get('podcast', {}).get('title')
-        title = ('%s: %s' % (podcast_title, episode_title)
-                 if podcast_title else episode_title)
+        blocks = traverse_obj(art_json, (
+            'sprinkledBody', 'content', ..., ('ledeMedia', None),
+            lambda _, v: v['__typename'] in ('Video', 'Audio')))
```

Comment:
I think we should raise here if we don't find anything. Better than an IndexError later imo
```suggestion
        blocks = traverse_obj(art_json, (
            'sprinkledBody', 'content', ..., ('ledeMedia', None),
            lambda _, v: v['__typename'] in ('Video', 'Audio')))
        if not blocks:
            raise ExtractorError('Unable to extract any media blocks from webpage')
```